### PR TITLE
Next: Forgot password support

### PR DESCRIPTION
**Pull request by carlokok**
### Short Description and Why It's Useful
This implements the forgot password sequence, by adding 3 things:

* Forgot password page in login model
* "we've sent you a mail" page in login model
* Landing page to enter the new password (/new-password/my%40example.com/token)

### Screenshots of Visual Changes before/after (if There Are Any)
afbeelding

afbeelding

afbeelding

### Which Environment This Relates To
* [x] Next version

### Upgrade Notes and Changelog
* [x] No upgrade steps required (100% backward compatibility and no breaking changes)

While this is a non breaking change, nothing happens if the useUser implementation doesn't actually implement the new methods. (There's no changelog/upgrade notes in Next yet)

### Contribution and Currently Important Rules Acceptance
* [x] I read and followed [contribution rules](https://github.com/DivanteLtd/vue-storefront/blob/master/CONTRIBUTING.md)

┆Issue is synchronized with this [Jira Story](https://vsf.atlassian.net/browse/NEXT-102) by [Unito](https://www.unito.io/learn-more)

### CODE REVIEW

**Comment by filrak** [comment]
While `forgotPassword` makes sense isn't `createPassword` just `updateUser` ?

**Comment by carlokok** [comment]
not exactly. Though I agree I don't like the name, I can't think of anything better. Think of this flow:

* User enters email & clicks forgot password
* Server backend sends a mail that ends up with a link like http://localhost:3000/new-password/me%40example.com/tokengeneratedbyserver)
* User enters a new password
* Given the token, the username and the new password, reset the password

So it does update the user, but the user isn't logged in at this point, so can't call update user.

**Comment by filrak** [comment]
ok that makes sense to me, @andrzejewsky wdyt?

File: packages/core/theme-module/theme/components/LoginModal.vue [hunk 1]
```diff
@@ -166,10 +212,11 @@ export default {
   },
   setup() {
     const form = ref({});
-    const isLogin = ref(false);
+    const isLogin = ref(1);
```
Comment (filrak) [inline packages/core/theme-module/theme/components/LoginModal.vue:215]: I think it's better to use strings here (or enums) to make it more readable

File: packages/core/theme-module/theme/components/LoginModal.vue [hunk 2]
```diff
@@ -6,7 +6,7 @@
       class="modal"
       @close="toggleLoginModal">
       <transition name="fade" mode="out-in">
-        <div v-if="isLogin" key="log-in">
+        <div v-if="isLogin == 1" key="log-in">
```
Comment (filrak) [inline packages/core/theme-module/theme/components/LoginModal.vue:9]: Wouldn't it be better to have these 3 options as separate components?

**Comment by filrak** [comment]
Also would you mind implementing the desired functionality you have putted into the interface. I think you need to do this anyway

File: packages/core/theme-module/theme/components/LoginModal.vue (hunk 1 above)
Comment (carlokok) [reply packages/core/theme-module/theme/components/LoginModal.vue]: Done (string keys)

**Comment by carlokok** [comment]
not quite sure what you meant with your last reply?

**Comment by bloodf** [comment]
This is closed due to inactivity.

If you want your PR to be re-opened, please update your files, check for any conflicts.

If your PR is for Vue Storefront 1 target the new repository at https://github.com/vuestorefront/vue-storefront-1

**Pull request closed without merging**